core/src/impl/Kokkos_ViewMapping.hpp

```diff
@@ -1771,6 +1767,23 @@ struct ViewOffset< Dimension , Kokkos::LayoutRight
       // Also requires equal static dimensions ...
     }
 
+  template< class DimRHS >
+  KOKKOS_INLINE_FUNCTION
+  ViewOffset( const ViewOffset< DimRHS , Kokkos::LayoutStride , void > & rhs )
+    : m_dim( rhs.m_dim.N0 , rhs.m_dim.N1 , rhs.m_dim.N2 , rhs.m_dim.N3
+           , rhs.m_dim.N4 , rhs.m_dim.N5 , rhs.m_dim.N6 , rhs.m_dim.N7 )
+    , m_stride( rhs.stride_0() )
+    {
+      if ( ((dimension_type::rank == 2)?rhs.m_stride.S1:
+           ((dimension_type::rank == 3)?rhs.m_stride.S2:
+           ((dimension_type::rank == 4)?rhs.m_stride.S3:
+           ((dimension_type::rank == 5)?rhs.m_stride.S4:
+           ((dimension_type::rank == 6)?rhs.m_stride.S5:
+           ((dimension_type::rank == 7)?rhs.m_stride.S6:rhs.m_stride.S7)))))) != 1 ){
+        Kokkos::abort("Kokkos::Impl::ViewOffset assignment of LayoutRight from LayoutStride requires right-most stride == 1" );
+      }
```

Comment:
I think we also need to check that the lower rank strides are the product of the base stride * dimensions up to that point. 

i.e. something like this:
```c++
stride[rank-1] == 1;
for(int i=rank-2; i>0;i++)
  stride[i] == rhs.extent[i+1]*stride[i+1];
stride[0] == rhs.stride_0() && stride[0] >= stride[1] * rhs.extent[1];
```

Reply:
Ok looks right.